app/objects/secondclass/c_fact.py

```diff
@@ -59,10 +95,17 @@ def __eq__(self, other):
             return self.unique == other.unique
         return False
 
-    def __init__(self, trait, value=None, score=1, collected_by=None, technique_id=None):
+    def __init__(self, trait, value=None, score=1, source=None, origin_type=None, links=None,
+                 relationships=None, limit_count=-1, collected_by=None, technique_id=None):
         super().__init__()
         self.trait = trait
         self.value = value
+        self.created = datetime.now()
         self.score = score
+        self.source = source
+        self.origin_type = origin_type
+        self.links = links or []
+        self.relationships = relationships or []
+        self.limit_count = limit_count
```

Comment:
Whats this field denoting specifically?

Reply:
It is supposed to serve the purpose of `restrictions` field. On my first couple of changes I had it as an enum following the initial design of a UNIQUE or SINGLE use for a Fact. After having a discussion with the team, it started to look like a counter referring to the amount of times this Fact could be re-used (hence the change to a Integer Field).

As far as naming goes I think you can find a bit of that discussion above on the Outdated comments.